silq/instrument_interfaces/keysight/E8267D_interface.py

```diff
@@ -363,13 +401,12 @@ def get_additional_pulses(self, interface: InstrumentInterface):
                             'input_channel': 'trig_in'})]
 
         if interface.IQ_modulation() == 'off':
-            if interface.frequency_deviation() == 0:  # No IQ modulation nor FM
+            if settings['frequency_deviation']() == 0:  # No IQ modulation nor FM
                 amplitude_FM = None
                 frequency_IQ = None
-                pass
             else:  # No IQ modulation, but FM
-                frequency_difference = self.pulse.frequency - interface.frequency()
-                amplitude_FM = frequency_difference / interface.frequency_deviation()
+                frequency_difference = self.pulse.frequency - settings['frequency']
+                amplitude_FM = frequency_difference / settings['frequency_deviation']
                 frequency_IQ = None
         else:  # interface.IQ_modulation() == 'on'
             if interface.FM_mode() == 'ramp':
```

Comment:
what's the advantage of substituting interface by settings only in some cases and not in all like interface.FM_mode()?